docs/guide/request_lifecycle.md

```diff
@@ -0,0 +1,225 @@
+# Request lifecycle: from request to response
+
+
+
+## HTTP request entering
+
+When entering the architecture, the HTTP request can be handled by several component like a firewall, a load balancer, a reverse-proxy, etc. before arriving on the web server itself.
+
+For an overview of what happens on reverse proxy like Varnish or Fastly, see [Context-aware HTTP cache / Request lifecycle](cache/context_aware_cache/#request-lifecycle).
+
+When arriving at a Web server, the request is filtered by Apache Virtual Host, Nginx Server Blocks or equivalent. This mainly (and shortly) separates requests of static resources from requests to the application.
+
+As Ibexa DXP is a Symfony application, everything starts like in Symfony, see [Symfony and HTTP Fundamentals](https://symfony.com/doc/current/introduction/http_fundamentals.html).
+
+If the HTTP request is to be treated by Ibexa DXP, it goes to the [Symfony Front Controller](https://symfony.com/doc/current/configuration/front_controllers_and_kernel.html#the-front-controller) public/index.php
+
+The Front Controller transform the HTTP request into a PHP [`Request` object](https://symfony.com/doc/current/introduction/http_fundamentals.html#symfony-request-object) and pass it to Symfony's Kernel to obtain a [`Response` object](https://symfony.com/doc/current/introduction/http_fundamentals.html#symfony-response-object) that will be transform and sent back as an HTTP response.
+
+The schemas start with a `Request` object entering Symfony and Ibexa DXP, only regular request from a browser. No ESI, no REST nor GraphQL request.
+
+
+
+## Lifecycle flowcharts
+
+### Concept flowchart
+
+The chart below introduces the logic of the request treatment.
+
+![Simplified request lifecycle flowchart](img/request_lifecycle_concept.png)
+
+The next chart will detail how this is implemented using Symfony HTTP kernel listeners.
+
+### Kernel events flowchart
+
+The following chart shows events, listeners and attributes added to the request or its wrapping event object.
+
+<img src="../img/request_lifecycle_events.png" width="741" height="2061" style="max-height: none;" alt="Detailed request lifecycle flowchart organised around kernel events" />
```

Comment:
We could actually use markdown here if we make use of one of the plugins available for mkdocs.
This would require adding to mkdocs.yml:
```
markdown_extensions:
  - attr_list
 ```

Then the following should work:
```
![Detailed request lifecycle flowchart organised around kernel events](img/request_lifecycle_events.png){: width="741" height="2061" style="max-height: none;"}
```

Reply:
Done in d4376faa5d676e4ef718ec054671467a3a3c0af7